### ISSUE TITLE
Discussion on pluggable discovery

### ISSUE THREAD

**Issue by oscargdd** [ietf-ccamp-wg/draft-ietf-ccamp-actn-poi-pluggable-usecases-gaps#9]
- Can the pluggable have "default values"? (e.g. application code, frequency...)

**Comment by oscargdd** [comment]
Should we add a sentence also meaning that the plugable must start in shutdown mode?

**Comment by ggrammel** [comment]
OIF-400ZR-010_reduced2.pdf defines for 400ZR, DWDM amplified - Application Code (0x01) a default Channel frequency 193.7THz which suggests it comes pre-configured out of the factory. If such pre-configuration is implemented, it should be learned during discovery phase.

**Comment by ju7ien** [comment]
> The play will come into picture with (1) pre-config from NE or (2) config after plug. The replacement of pluggable falls under (1).

I agree with both scenarios. It may be obvious, but it relies on the fact that shelf configuration has precedence on any factory default. However, those don't justify that we preclude default values: there are uses cases where the expectation is to only have to enable the interfaces (e.g., single wavelength on dark fiber).

**Comment by ggalimba56** [comment]
I'd add to Julien Comments: the default configuration will be applied in absence of port configuration. I.e. the Pluggable slot has never hosted a pluggable or a service has never configured to that port.

**Comment by oscargdd** [comment]
[Discovery_workflows.pptx](https://github.com/oscargdd/draft-poidt-ccamp-actn-poi-pluggable-usecases-gaps/files/14998752/Discovery_workflows.pptx)

These are the scenarios discusssed

**Comment by jbouqui153** [comment]
Here is our proposal with Manolo as agreed in our last call to be added in "Coherent DWDM pluggable insertion in the router linecard port ('valid for coherent pluggable')" regarding the default parameters topic in pluggables. When I can I will do the Pull Request. Comments are welcome.

"Several coherent pluggables seem to come with a factory default set of parameters (e.g. default channel number, default launched power, default application code id, laser-on, admin-state enabled etc.). This factory default set of parameters varies from manufacturer to manufacturer. This can allow a “plug&play” mode of operation over point-to-point connections (e.g. single wavelength over dark fiber) of course limited to bookended solutions from the same Vendor and same factory default parameters. However, when the optical connection between two pluggables is targeted to run over a DWDM  Open Line System (OLS) network, optical validation & planning step is first required to determine the right target parameters values to be set in the pluggables before interconnecting them to their respective ROADM to avoid to impact any other existing optical channels already up and running in the OLS network.
It is critical for operators to have the same kind of commissioning phase independently of the deployment scenario: point-to-point vs ROADM meshed OLS network. As a consequence, the use of default parameters may be fine but the default set of factory parameters shall be able to be overwritten through router CLI or through Packet PNC to another set of operational default parameters defined by the operator that will change from pluggable to pluggable when deployed over an OLS network. A reset of the coherent pluggable (through router CLI or through Packet PNC) shall always go back to this operator’s default set of operational parameters where, for example, the laser-state shall be ‘Off’ and admin-state ‘disabled’."

**Comment by oscargdd** [comment]
Thanks for the proposal. Comments below:

1) The ·pre-configuration can come from factory, or operator-driven. E.g. you want to pre-tune your pluggables, etc... It might be useful in some scenarios.  

Text suggestion:

"Coherent pluggables can come with a default set of parameters, either by factory settings or pre-configured by the operator. Pre-configured parameters might include default channel number, default launched power, default application code id, laser-on, admin-state enabled etc.."

2) The fact that it varies is because there is no "enforcement" by any standard right? I also understood that CMIS defines (or maybe recommends...) some defaul parameters. 

Text suggestoin:

"This factory default set of parameters varies from manufacturer to manufacturer, as the standard does not mandate a given set of values, even though some parameters have default values."

3) I would add that the recommendation is that, independently wethere the pluggable comes with a valid pre-configuration or not, it should not inmideatly start transmitting.

Text suggestion:

"In order to avoid unintentionally harming other channels, it is recommended that after a pluggable is connected to the router, by default, the laser is off, and the operator (or the controller) should explictly enable the transceiver."

4) "overwritten through router CLI or through Packet PNC". I think we need to hightlight the the router can have also been preconfigured, and it MUST take prececence over the pluggable. If you modify later, it is a normal case...  Also, I would be agnostic of who configured that (maybe saying management or control is fine) I would suggest:

"shall be overwritten throught the configuration stored in the router, which might have been provided by management or controller means. "


When you make the PR, I make the suggestions there directly, we can continue the discussion here

**Comment by oscargdd** [comment]
Solved with Jeff pull requests.
